src/MoBi.Presentation/Presenter/SelectReferencePresenterBase.cs

```diff
@@ -241,7 +241,14 @@ private void selectInView(IEntity entityToSelect)
       protected void AddSpatialStructures()
       {
          var spatialStructures = _buildingBlockRepository.SpatialStructureCollection;
-         _view.AddNodes(spatialStructures.MapAllUsing(_referenceMapper));
+         var nodes = spatialStructures.Select(x =>
+         {
+            var mapped = _referenceMapper.MapFrom(x);
+            mapped.Text = x.DisplayName;
+            return mapped;
+         });
```

Comment:
I didn't realize as we discussed this together.

There's an extension method called `WithText` and you can chain it. That would make this LINQ into

`var nodes = spatialStructures.Select(x => _referenceMapper.MapFrom(x).WithText(x.DisplayName));`

Which could also be inserted directly into `AddNodes` instead of declaring a variable (if you think it's still readable).